cmake/build_env_info.cmake

```diff
@@ -106,7 +106,7 @@ function(check_git_setup)
   )
 
   add_library(impl_git_version ${CMAKE_CURRENT_BINARY_DIR}/generated/Kokkos_Version_Info.cpp)
-  target_include_directories(impl_git_version PUBLIC ${CMAKE_BINARY_DIR}/generated)
+  target_include_directories(impl_git_version PUBLIC ${CMAKE_CURRENT_BINARY_DIR}/generated)
```

Comment:
I see.  So it was an oversight in #7557 
We should consider cherry-picking into 4.6.1 since we haven't released yet